plugins/mentions/plugin.js

```diff
@@ -0,0 +1,448 @@
+		this.minChars = config.minChars !== null && config.minChars !== undefined ? config.minChars : MIN_CHARS;
+
+		/**
+		 * Pattern used to match queries.
+		 *
+		 * Default pattern matches words with query including {@link #marker marker} and {@link #minChars minChars} properties.
+		 *
+		 * ```javascript
+		 * // Match only words starting with "a".
+		 * var pattern = /^a+\w*$/;
+		 * ```
+		 *
+		 * @property {RegExp} pattern
+		 * @readonly
+		 */
+		this.pattern = config.pattern || createPattern( this.marker, this.minChars );
+
+		/**
+		 * Indicates if URL feed responses will be cached.
+		 *
+		 * The cache is based on URL request and shared between all mentions instances (including different editors).
+		 *
+		 * @property {Boolean} [cache=true]
+		 * @readonly
+		 */
+		this.cache = config.cache !== undefined ? config.cache : true;
+
+		/**
+		 * @inheritdoc CKEDITOR.plugins.mentions.configDefinition#throttle
+		 * @property {Number} [throttle]
+		 * @readonly
+		 */
+		this.throttle = config.throttle !== undefined ? config.throttle : 200;
+
+		/**
+		 * {@link CKEDITOR.plugins.autocomplete Autocomplete} instance used by mentions feature to implement autocompletion logic.
+		 *
+		 * @property {CKEDITOR.plugins.autocomplete}
+		 * @private
+		 */
+		this._autocomplete = new CKEDITOR.plugins.autocomplete( editor, {
+			textTestCallback: getTextTestCallback( this.marker, this.minChars, this.pattern ),
+			dataCallback: getDataCallback( feed, this ),
+			itemTemplate: config.itemTemplate,
+			outputTemplate: config.outputTemplate,
+			throttle: this.throttle,
+			itemsLimit: config.itemsLimit
+		} );
+	}
+
+	Mentions.prototype = {
+
+		/**
+		 * Destroys the mentions instance.
+		 *
+		 * View element and event listeners will be removed from the DOM.
+		 */
+		destroy: function() {
+			this._autocomplete.destroy();
+		}
+	};
+
+	function createPattern( marker, minChars ) {
+		var pattern = '\\' + marker + '\\w';
+
+		if ( minChars ) {
+			pattern += '{' + minChars + ',}';
+		} else {
+			pattern += '*';
+		}
+
+		pattern += '$';
+
+		return new RegExp( pattern );
+	}
+
+	function getTextTestCallback( marker, minChars, pattern ) {
+		return function( range ) {
+			if ( !range.collapsed ) {
+				return null;
+			}
+
+			return CKEDITOR.plugins.textMatch.match( range, matchCallback );
+		};
+
+		function matchCallback( text, offset ) {
+			var match = text.slice( 0, offset )
+				.match( pattern );
+
+			if ( !match ) {
+				return null;
+			}
+
+			// Do not proceed if a query is a part of word.
+			var prevChar = text[ match.index - 1];
+			if ( prevChar !== undefined && !prevChar.match( /\s+/ ) ) {
+				return null;
+			}
+
+			return {
+				start: match.index,
+				end: offset
+			};
+		}
+	}
+
+	function getDataCallback( feed, mentions ) {
+		return function( query, range, callback ) {
+			// We are removing marker here to give clean query result for the endpoint callback.
+			if ( mentions.marker ) {
+				query = query.substring( 1 );
```

Comment:
The length of a marker should be dynamically computed, so `mentions.marker.length` instead of magic number 1